### PR TITLE
Autocomplete: fix XSS in JSONP demo

### PR DESCRIPTION
**Pull request by wenz**
fixes #15048

### CODE REVIEW

File: demos/autocomplete/search.php [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 sleep( 2 );
 // no term passed - just exit early with no response
-if (empty($_GET['term'])) exit ;
+if (!isset($_GET["term"]) || empty($_GET['term'])) exit ;
```
Comment (scottgonzalez) [inline demos/autocomplete/search.php]: This `isset()` is duplicative of the `empty()` check.

Comment (wenz) [inline demos/autocomplete/search.php]: agreed. Fixed.

**Comment by scottgonzalez** [comment]
This is too restrictive; it would reject many valid callback function names. Properly validating the callback name is too complex and there's no actual attack vector to worry about here.

**Comment by wenz** [comment]
I respectfully disagree.

1) in the context of the demo, the filtering works fine, since it alows jQuery's default callback name consisting of jQuery, many digits, and one underscore. In addition, which valid callback function name is rejected by the regex? We might consider making $ and . valid characters, but what else is missing?
2) PoC: put the (unpatched) file on a web server and load http://server/path/to/search.php?callback=<script>alert(1)</script>&term=X in a browser without built-in XSS protection (e.g. Firefox). The JavaScript code is executed in the security context of the target server, a classic XSS vector.

**Comment by scottgonzalez** [comment]
> In addition, which valid callback function name is rejected by the regex? We might consider making $ and . valid characters, but what else is missing?

Tons and tons of unicode characters are missing.

> 2) PoC: put the (unpatched) file on a web server and load http://server/path/to/search.php?callback=&term=X in a browser without built-in XSS protection (e.g. Firefox). The JavaScript code is executed in the security context of the target server, a classic XSS vector.

The JavaScript code being an array, which does nothing. How is that a classic XSS vector?

**Comment by wenz** [comment]
GitHub ate the script tag in my URL. Try this:
http://server/path/to/search.php?callback=[script]alert(1)[/script]&term=X
but replace the square brackets with angle brackets.

**Comment by scottgonzalez** [comment]
So this is about linking directly to `search.php` and injecting HTML, not about injecting JS in the JSONP request. Simply passing the callback through `htmlspecialchars()` would prevent someone from pretending to load a real page and won't break any valid callback names.

**Comment by wenz** [comment]
htmlspecialchars() would also work, however I personally prefer the validation approach over escaping here, since a callback name with (for instance) angle brackets should lead to nothing being returned. 

The proposed fix works for the sample code. If someone is changing the example, e.g. by using a callback name with unicode characters, it's probably their responsibility to adapt the PHP script, as well. If you wish, I could amend the PR to use htmlspecialchars() despite my reservations. 

Anyway, this must be fixed asap, since the same file is also residing on the jqueryui.com domain.

**Pull request closed without merging**